python/ray/train/lightning/lightning_trainer.py

```diff
@@ -142,14 +144,19 @@ def fit_params(self, **kwargs) -> "LightningConfigBuilder":
         self._trainer_fit_params.update(**kwargs)
         return self
 
-    def ddp_strategy(self, **kwargs) -> "LightningConfigBuilder":
+    def strategy(self, name, **kwargs) -> "LightningConfigBuilder":
         """Set up the configurations of ``pytorch_lightning.strategies.DDPStrategy``.
 
         Args:
+            name: The name of your distributed strategy. You can choose
+                from "ddp" and "fsdp". Default: "ddp".
             kwargs: For valid arguments to pass, please refer to:
                 https://lightning.ai/docs/pytorch/stable/api/lightning.pytorch.strategies.DDPStrategy.html
+                and
+                https://lightning.ai/docs/pytorch/stable/api/lightning.pytorch.strategies.FSDPStrategy.html
         """
-        self._ddp_strategy_config.update(**kwargs)
+        self._strategy_config["_strategy_name"] = name
+        self._strategy_config.update(**kwargs)
```

Comment:
Can we throw an error if the name they pass is not in the allowed set? And tell them what is allowed in the error message. + a small unit test for that.

Reply:
Ah, I see you do it later. I think we should just throw on this call to `.strategy`, rather than on build.